packages/compiler-sfc/src/compileScript.ts

```diff
@@ -1889,17 +1890,22 @@ function extractRuntimeProps(
   const members = node.type === 'TSTypeLiteral' ? node.members : node.body
   for (const m of members) {
     if (
-      (m.type === 'TSPropertySignature' || m.type === 'TSMethodSignature') &&
-      m.key.type === 'Identifier'
+        (m.type === 'TSPropertySignature' || m.type === 'TSMethodSignature') &&
+        (m.key.type === 'Identifier' || m.key.type === 'StringLiteral')
     ) {
-      let type
+      let type, keyName;
+      if(m.key.type === 'StringLiteral'){
+        keyName = `"${m.key.value}"`
+      } else {
+        keyName = m.key.name
+      }
       if (m.type === 'TSMethodSignature') {
         type = ['Function']
       } else if (m.typeAnnotation) {
         type = inferRuntimeType(m.typeAnnotation.typeAnnotation, declaredTypes)
       }
-      props[m.key.name] = {
-        key: m.key.name,
+      props[keyName] = {
+        key: keyName,
         required: !m.optional,
         type: type || [`null`]
       }
```

Comment:
I think we can store the raw name into the object, instead of re-get it through splitting string.

Reply:
```suggestion
      props[keyName] = { // no quotes
        key: keyName,
        required: !m.optional,
        type: type || [`null`],
        rawName: nameWithoutQuotes // m.key.extra.raw
      }
```